workspaces/ballerina/type-diagram/src/components/entity-relationship/EntityNode/EntityHead/EntityHead.tsx

```diff
@@ -106,6 +113,14 @@ export function EntityHeadWidget(props: ServiceHeadProps) {
         setAnchorEl(null);
     };
 
+    const onNodeDeleteWithConfirm = async () => {
+        // Proceed with deletion only if no links exist
+        if (node?.entityObject?.editable && onNodeDelete) {
```

Comment:
lets remove unrelated comments. since with the current logic we check delete support based on LS